chia/wallet/did_wallet/did_wallet.py

```diff
@@ -1266,3 +1249,63 @@ async def save_info(self, did_info: DIDInfo, in_transaction: bool):
         wallet_info = WalletInfo(current_info.id, current_info.name, current_info.type, data_str)
         self.wallet_info = wallet_info
         await self.wallet_state_manager.user_store.update_wallet(wallet_info, in_transaction)
+
+    def generate_wallet_name(self) -> str:
+        """
+        Generate a new DID wallet name
+        :return: wallet name
+        """
+        max_num = 0
+        for wallet in self.wallet_state_manager.wallets.values():
+            if wallet.type() == WalletType.DISTRIBUTED_ID:
+                matched = re.search(r"^Profile (\d+)$", wallet.wallet_info.name)
+                if matched and int(matched.group(1)) > max_num:
+                    max_num = int(matched.group(1))
+        return f"Profile {max_num + 1}"
+
+    def check_existed_did(self):
+        """
+        Check if the current DID is existed
+        :return: None
+        """
+        for wallet in self.wallet_state_manager.wallets.values():
+            if (
+                wallet.type() == WalletType.DISTRIBUTED_ID
+                and self.did_info.origin_coin.name() == wallet.did_info.origin_coin.name()
+            ):
+                self.log.warning(f"DID {self.did_info.origin_coin} already existed, ignore the wallet creation.")
+                raise ValueError("Wallet already exists")
+
+    @staticmethod
+    def get_did_from_file(filename: str) -> DIDInfo:
```

Comment:
I don't think that we should be reading/writing files in the wallet.  We have this ability to run remote machines and connect GUIs to them from a computer that is not actually running the wallet.  The wallet and all other services should always ship a blob across the RPC and then let the UX write it to a file if it wants.  Similarly it should take a blob from the RPC rather than reading it from a file.

Reply:
So I think this should also be true for creating the attestments?